core/sdk/src/main/java/com/mongodb/stitch/core/auth/internal/CoreStitchAuth.java

```diff
@@ -249,7 +249,12 @@ private synchronized Response doAuthenticatedRequest(
                                       final Decoder<T> resultDecoder) {
     final Response response = doAuthenticatedRequest(stitchReq);
     try {
-      return BsonUtils.parseValue(IoUtils.readAllToString(response.getBody()), resultDecoder);
+      final String bodyStr = IoUtils.readAllToString(response.getBody());
+      if (bodyStr.equals("null")) {
```

Comment:
Please leave a comment here for why we do this. It's a very odd inclusion without explanation.